src/test/resources/recipes/ShouldAddImportsRecipes.java

```diff
@@ -100,7 +106,47 @@ public J visitMethodInvocation(J.MethodInvocation elem, ExecutionContext ctx) {
                 }
 
             };
+            return Preconditions.check(
+                    new UsesType<>("java.util.Objects", false),
```

Comment:
Out of curiosity: Why is there no `UsesMethod` for `Objects#equals()` here? I would have expected the precondition to check for either `Objects#equals()` or `Integer#compare()`.

Reply:
Currently those would only be generated if we use a static import of those methods; we can expand that of course, but figured this first implementation already limits the execution of the recipes to recipes using the type, which at scale already trims it down quite a bit.

Reply:
Thanks for the explanation. Yes, also covering the others would indeed make sense.

Reply:
Logged as an improvement:
- https://github.com/openrewrite/rewrite-templating/issues/26